source/Calamari/Kubernetes/ResourceStatus/CountdownTimer.cs

```diff
@@ -31,7 +31,7 @@ public CountdownTimer(TimeSpan duration)
         public void Start() => stopwatch.Start();
         public void Reset() => stopwatch.Reset();
         public bool HasStarted() => stopwatch.IsRunning;
-        public bool HasCompleted() => stopwatch.Elapsed >= duration;
+        public bool HasCompleted() => HasStarted() && stopwatch.Elapsed >= duration;
```

Comment:
This could be fixed with just changing `>=` to `>`, but I think this way communicate the intention better. A timer must be started before it completes.